src/memex/storage.py

```diff
@@ -108,14 +109,14 @@ def create_annotation(request, data):
             )
 
     # The user must have permission to create an annotation in the group
-    # they've asked to create one in.
-    if data['groupid'] != '__world__':
-        group_principal = 'group:{}'.format(data['groupid'])
-        if group_principal not in request.effective_principals:
-            raise schemas.ValidationError('group: ' +
-                                          _('You may not create annotations '
-                                            'in groups you are not a member '
-                                            'of!'))
+    # they've asked to create one in. If the application didn't configure
+    # a groupfinder we will allow writing this annotation without any
+    # further checks.
+    group = groups.find(request, data['groupid'])
```

Comment:
As discussed in Slack, this is due to [this](https://github.com/hypothesis/h/blob/master/h/resources.py#L15) when the authenticated user is an admin, and that Pyramid uses the request context when the passed-in one is `None`.
The easiest fix for now is to check for `None` as well as `has_permission`.

Fixed